external-import/recorded-future/src/rflib/rf_alerts.py

```diff
@@ -1,389 +1,441 @@
-            "CONNECTOR_INITIAL_PULL_OFF",
-            ["alert", "initial_pull_off"],
-            config,
-            required=False,
-            default=1,
-        )
-        self.connector_download_path = pycti.get_config_variable(
-            "CONNECTOR_DOWNLOAD_PATH", ["alert", "download_path"], config, required=True
-        )
-        self.recordedfuture_logo_abuse_rule_id = pycti.get_config_variable(
-            "RECORDED_FUTURE_LOGO_ABUSE_RULE_ID",
-            ["alert", "logo_abuse_rule_id"],
-            config,
-            required=True,
-        )
-
-        self.opencti_api_client = pycti.OpenCTIApiClient(
-            url=self.opencti_url, token=self.opencti_token, ssl_verify=True
+        self.api_recorded_future = RecordedFutureApiClient(
+            x_rf_token=rf_token,
+            helper=helper,
+            base_url="https://api.recordedfuture.com/",
         )
 
-        # Querying incident_type_ov vocabularies
-        self.vocabulary_list = []
-        existing_vocabulary = self.opencti_api_client.vocabulary.list(
+        self.stix_channels = self.helper.api.vocabulary.list(
             **{
                 "filters": {
                     "mode": "and",
                     "filterGroups": [],
                     "filters": [
                         {
                             "key": "category",
-                            "values": self.opencti_incident_vocabulary_name,
+                            "values": "channel_types_ov",
                         }
                     ],
                 }
             }
         )
-        for vocab in existing_vocabulary:
-            self.vocabulary_list.append(
-                Vocabulary(vocabulary_id=vocab["id"], vocabulary_name=vocab["name"])
-            )
 
-        self.api_recorded_future = RecordedFutureApiClient(
-            x_rf_token=self.recordedfuture_x_rf_token,
-            connector_download_path=self.connector_download_path,
+    @staticmethod
+    def _create_author():
+        """Creates Recorded Future Author"""
+        return stix2.Identity(
+            id=pycti.Identity.generate_id("Recorded Future", "organization"),
+            name="Recorded Future",
+            identity_class="organization",
         )
-        self.api_recorded_future.get_prioritedrule_ids(limit=100)
-        self.api_recorded_future.compare_rf_rules_and_vocabularies(self.vocabulary_list)
 
-        for newrule in self.api_recorded_future.unfound_rf_rules_in_vocabularies:
-            self.opencti_api_client.vocabulary.create(
-                **{
-                    "name": str(
-                        self.api_recorded_future.nomenclature + newrule.rule_name
-                    ),
-                    "description": "Vocabulary created by a new priorited rule in RecordedFuture.",
-                    "category": self.opencti_incident_vocabulary_name,
-                }
-            )
+    def get_root_domain(self, url):
```

Comment:
Is this code useful? To delete and by the same way, delete the "tldextract" dependency ?

Reply:
Will remove it, it is not used